src/Elastic.Apm/Report/PayloadSenderV2.cs

```diff
@@ -199,9 +205,10 @@ private async Task ProcessQueueItems(object[] queueItems)
 		{
 			try
 			{
-				var metadataJson = _payloadItemSerializer.SerializeObject(_metadata);
 				var ndjson = new StringBuilder();
-				ndjson.AppendLine("{\"metadata\": " + metadataJson + "}");
+				if (_cachedMetadataJsonLine == null)
```

Comment:
Unfortunately it was not that straightforward - I tried to generate `_cachedMetadataJsonLine`  in `PayloadSenderV2`'s constructor (IMHO if possible immutable state is preferable to mutable) but it didn't work because `_metadata` depends on `service` passed to `PayloadSenderV2`'s constructor but `service` is changed after the call `PayloadSenderV2`'s constructor and we need that changed value in `_cachedMetadataJsonLine`. I think it's a problematic point when initialization order is not sequential and it might cause bugs later.